File: backend/routes/auth_routes.py
from flask import Blueprint, request, jsonify
from services.user_service import register_customer, login_user

auth_bp = Blueprint('auth', __name__)

@auth_bp.post("/register")
def register():
    data = request.get_json()
    try:
        response = register_customer(data)
        return jsonify(response), 201
    
    except ValueError as e:
        return jsonify({"error": str(e)}), 400
    
    except Exception as e:
        return jsonify({"error": "Registration failed", "details": str(e)}), 500


@auth_bp.post("/login")
def login():
    data = request.get_json()
    email = data.get("email")
    password = data.get("password")

    if not email or not password:
        return jsonify({"error": "Email and password are required."}), 400
    
    response, status = login_user(email, password)
    return jsonify(response), status

File: backend/routes/auth_routes.js
const express = require('express');
const router = express.Router();
const { registerCustomer, loginUser } = require('../services/user_service');

router.post("/register", async (req, res) => {
  try {
    const response = await registerCustomer(req.body);
    return res.status(201).json(response);
  } catch (error) {
    if (error.message === "Email already registered") {
      return res.status(400).json({ error: error.message });
    }
    return res.status(500).json({ error: "Registration failed", details: error.message });
  }
});

router.post("/login", async (req, res) => {
  const { email, password } = req.body;
  
  if (!email || !password) {
    return res.status(400).json({ error: "Email and password are required." });
  }
  
  const [response, status] = await loginUser(email, password);
  return res.status(status).json(response);
});

module.exports = router;
